Actor.cs

```diff
@@ -0,0 +1,85 @@
+internal class Actor : MonoBehaviour
+{
+	private static List<Actor> _actors = new List<Actor>();
+	public int StartingHealth;
+	int _currentHealth;
+	private bool _isDead;
+	private CancellationTokenSource _healthCancellationTokenSource;
+	private CancellationTokenSource _deathCancellationTokenSource;
+
+	public event Action<Actor> ActorDied;
+	public event Action<Actor> ActorHealthChanged;
+
+	public void TakeDamage(int damage)
+	{
+		_currentHealth -= damage;
+		ActorHealthChanged?.Invoke(this);
+		RecordHealthToServer();
+		
+		if (_currentHealth <= 0)
+		{
+			_isDead = true;
+			ActorDied?.Invoke(this);
+			RecordDeathToServer();
+		}
+	}
+
+	private void Start()
+	{
+		_currentHealth = StartingHealth;
+	}
+
+	private void OnEnable()
+	{
+		_actors.Add(this);
+	}
+
+	void RecordHealthToServer()
+	{
+		if (_healthCancellationTokenSource != null)
+		{
+			_healthCancellationTokenSource.Cancel();
+		}
+
+		_healthCancellationTokenSource = new CancellationTokenSource();
+		RecordHealthToServer(_healthCancellationTokenSource.Token).Forget();
```

Comment:
I believe `.Forget()` is a UniTask-ism not a c#-ism
We likely need to change this to be the normal c# task code which I think is the following
`Task.Run(() => RecordHealthTOServer(_healthCancellationTokenSource.Token));`

Also currently this'll not compile, you're missing the `Async` part of the method name